source/common/filesystem/BUILD

```diff
@@ -3,11 +3,39 @@ licenses(["notice"])  # Apache 2
 load(
     "//bazel:envoy_build_system.bzl",
     "envoy_cc_library",
+    "envoy_cc_platform_dep",
     "envoy_package",
 )
 
 envoy_package()
 
+envoy_cc_library(
+    name = "directory_lib",
+    hdrs = ["directory.h"],
+    deps = envoy_cc_platform_dep("directory_iterator_impl_lib"),
+)
+
+envoy_cc_library(
```

Comment:
I think what you need to do to make this work with `/...` paths is have a `envoy_cc_posix_library` and `envoy_cc_windows_library` as well that effectively nops these when the platform doesn't match the build host. That should fix CI.